status-reporter/status_reporter/rabbitmq_reporter.py

```diff
@@ -13,17 +13,103 @@
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
 
-from cloudify.cluster_status import CloudifyNodeType
 
-from .status_reporter import Reporter
+from requests import get
+from requests.exceptions import RequestException
+
+from cloudify.cluster_status import (CloudifyNodeType, ServiceStatus,
+                                     NodeServiceStatus)
+
+from .status_reporter import Reporter, logger
+from .constants import STATUS_REPORTER_CONFIG_KEY
+from .utils import get_systemd_services, get_node_status
+
+NODES = 'nodes'
+VHOSTS = 'vhosts'
+EXTRA_INFO = 'extra_info'
+NODE_STATUS = 'node_status'
+CLUSTER_STATUS = 'cluster_status'
+RABBITMQ_SERVICE_KEY = 'RabbitMQ'
+HEALTH_CHECK = 'healthchecks/node'
+CA_PATH = '/etc/cloudify/ssl/rabbitmq-ca.pem'
+RABBITMQ_URL = 'https://localhost:15671/api/'
+RABBITMQ_SERVICES = {'cloudify-rabbitmq.service': RABBITMQ_SERVICE_KEY}
 
 
 class RabbitMQReporter(Reporter):
     def __init__(self):
         super(RabbitMQReporter, self).__init__(CloudifyNodeType.BROKER)
 
     def _collect_status(self):
-        return {}
+        services, statuses = get_systemd_services(RABBITMQ_SERVICES)
+        extra_config = self._config.get(STATUS_REPORTER_CONFIG_KEY)
+        if self._rabbitmq_service_not_running(statuses) or not extra_config:
+            return self._rabbitmq_status_failed(services)
+        self._update_node_status(services, statuses, extra_config)
+        self._update_cluster_status(services, statuses, extra_config)
+        status = get_node_status(statuses)
+        return status, services
+
+    @staticmethod
+    def _rabbitmq_service_not_running(statuses):
+        return NodeServiceStatus.INACTIVE in statuses
+
+    def _update_cluster_status(self, services, statuses, config):
+        cluster_status, cluster_extra_info = self._get_cluster_status(config)
+        services[RABBITMQ_SERVICE_KEY][EXTRA_INFO][CLUSTER_STATUS] = \
+            cluster_extra_info
+        statuses.append(cluster_status)
+
+    def _get_cluster_status(self, config):
+        nodes_response = self._query_rabbitmq(config, NODES)
+        if not nodes_response:
+            return NodeServiceStatus.INACTIVE, {}
+        number_of_nodes = len(nodes_response.json())
```

Comment:
well you don't need this, because you can do len(running_nodes) no?

Reply:
Running nodes only shows the current running nodes, which means that if one is down (in a 3 nodes cluster), then there will be only 2 out of 3 nodes running. So we need this to make the comparison.